src/raw_branch.rs

```diff
@@ -347,8 +347,12 @@ where
     }
 
     pub fn leaf(&self) -> Option<&C::Leaf> {
-        if let Some(last) = self.levels.last() {
-            last.leaf()
+        if self.exact {
+            if let Some(last) = self.levels.last() {
+                last.leaf()
+            } else {
+                None
+            }
         } else {
             None
         }
```

Comment:
I think you might compact the above line as following:
```rust
if self.exact {
    self.levels.last()?.leaf()
} else {
    None
}
```

Reply:
That's much nicer yeah. fixed